src/include/packing_inline.h

```diff
@@ -754,3 +754,497 @@ __wt_struct_size_adjust(WT_SESSION_IMPL *session, size_t *sizep)
 
     *sizep = curr_size;
 }
+
+/*
+ * __wt_struct_size_system_record --
+ *     Calculate the system log rectype size of a packed byte string of format I.
+ */
+static inline void
+__wt_struct_size_system_record(WT_SESSION_IMPL *session, size_t *sizep, uint32_t rectype)
```

Comment:
Code layering wise this naming isn't great. The packing routines are designed to give a generic packing/unpacking interface, and describing how a specific sub-system uses that packing in the name violates that intention.

I know these started out as more generic type based and switched based on a review request. I agree that it would be nice for the caller to encapsulate functionality here, but in this case the relationship is backward. This is a utility function - it is important that there isn't knowledge about the caller pushed down (or the appearance of such).